docs/source/search-runs.rst

```diff
@@ -3,207 +3,240 @@
+
+  for i in range(10):
+      with mlflow.start_run():
+          mlflow.log_metrics(
+              {
+                  "loss": loss[i],
+                  "accuracy": accuracy[i],
+                  "log-scale-loss": log_scale_loss[i],
+                  "f1 score": f1_score[i],
+              }
+          )
+
+          mlflow.log_params(
+              {
+                  "batch_size": batch_size[i],
+                  "learning rate": learning_rate[i],
+                  "model": model[i],
+              }
+          )
+
+          mlflow.set_tags(
+              {
+                  "task": task[i],
+                  "environment": environment[i],
+              }
+          )
+
+          dataset = mlflow.data.from_numpy(
+              features=np.random.uniform(size=[20, 28, 28, 3]),
+              targets=np.random.randint(0, 10, size=[20]),
+              name=dataset_name[i],
+              digest=dataset_digest[i],
+          )
+          mlflow.log_input(dataset, context=dataset_context[i])
+
+
+The code above creates 10 MLflow runs with different metrics, params, tags and dataset information. After execution,
+you should find all these runs under the experiment "/search-run-guide", as shown by the screenshot:
+
+.. figure:: _static/images/search-runs/created_mlflow_runs.png
+   :alt: testing runs
+   :width: 90%
+   :align: center
 
-  .. code-block:: sql
 
-    params.alpha = "0.3" and params.lambda = "0.001" and metrics.error <= 0.05
+.. _search-runs-syntax:
 
+Search Query Syntax
+^^^^^^^^^^^^^^^^^^^
 
-Identifier
-^^^^^^^^^^
+You need to write search queries to fetch your MLflow runs. The query must be a condition or a few conditions
+concatenated by ``and``. At a high level, the syntax is similar to SQL WHERE clause without supporting ``or`` keyword.
+For example, below is a query to search for all runs that have logged a ``loss`` metric smaller than ``1`` to
+MLflow.
 
-Required in the LHS of a search expression. Signifies an entity to compare against.
+.. code-block:: sql
 
-An identifier has two parts separated by a period: the type of the entity and the name of the entity. The type of the entity is ``metrics``, ``params``, ``attributes``, ``datasets``, or ``tags``. The entity name can contain alphanumeric characters and special characters.
+  metric.loss < 1
 
-This section describes supported entity names and how to specify such names in search expressions.
+Briefly a search condition consists of 3 parts:
 
-.. contents:: In this section:
-  :local:
-  :depth: 1
+- An MLflow field on the left side, which can be some metric, param, tag, dataset or MLflow run metadata, e.g.,
+  ``metric.loss`` or ``param.batch_size``.
+- A comparator in the middle, we support 3 types of comparators:
+  - Numeric comparators: ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+  - String comparators: ``=``, ``!=``, ``LIKE`` and ``ILIKE``. String comparators are case sensitive.
+  - Set comparators: ``IN``.
```

Comment:
Let's also mention whether ``IN`` for string type is case-sensitive or case-insensitive. This is often confusing to user :)

Reply:
good call! 

Actually I did more testing, and `IN` only works for dataset information and run metadata search. I've updated the guide with the correpsonding information.